src/components/app/pagePolls/activePoll.tsx

```diff
@@ -0,0 +1,300 @@
+'use client'
+
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { differenceInDays } from 'date-fns'
+import Cookies from 'js-cookie'
+import { EyeIcon } from 'lucide-react'
+
+import { actionCreatePollVote } from '@/actions/actionCreatePollVote'
+import { LoginDialogWrapper } from '@/components/app/authentication/loginDialogWrapper'
+import { UserActionFormActionUnavailable } from '@/components/app/userActionFormCommon/actionUnavailable'
+import { CheckIcon } from '@/components/app/userActionGridCTAs/icons/checkIcon'
+import { Button } from '@/components/ui/button'
+import { Dialog, DialogBody, DialogContent, DialogTrigger } from '@/components/ui/dialog'
+import { Input } from '@/components/ui/input'
+import { Label } from '@/components/ui/label'
+import { PollResultsDataResponse, PollsVotesFromUserResponse } from '@/data/polls/getPollsData'
+import { useDialog } from '@/hooks/useDialog'
+import { USER_COUNTRY_CODE_COOKIE_NAME } from '@/utils/server/getCountryCode'
+import { SWCPoll } from '@/utils/shared/getSWCPolls'
+import { isValidCountryCode } from '@/utils/shared/isValidCountryCode'
+import { DEFAULT_SUPPORTED_COUNTRY_CODE } from '@/utils/shared/supportedCountries'
+import { cn } from '@/utils/web/cn'
+
+interface PollsProps {
+  activePoll: SWCPoll
+  isVoteAgain: boolean
+  handleShowResults: () => void
+  pollsResultsData: Record<string, PollResultsDataResponse>
+  userPollsData: PollsVotesFromUserResponse | undefined
+  isLoading: boolean
+}
+
+function SubmitButton({ isSubmitDisabled }: { isSubmitDisabled: boolean }) {
+  const dialogProps = useDialog({ analytics: 'Active Poll Submit Button' })
+
+  const userCountryCode = Cookies.get(USER_COUNTRY_CODE_COOKIE_NAME)
+  const isValid = isValidCountryCode({
+    countryCode: DEFAULT_SUPPORTED_COUNTRY_CODE,
+    userCountryCode,
+    bypassCountryCheck: false,
+  })
+
+  const setDialogOpen = (open: boolean) => {
+    dialogProps.onOpenChange(open)
+  }
+
+  return (
+    <LoginDialogWrapper
+      authenticatedContent={
+        isValid ? (
+          <Button disabled={isSubmitDisabled} size="lg" type="submit" variant="primary-cta">
+            Submit
+          </Button>
+        ) : (
+          <Dialog {...dialogProps} onOpenChange={setDialogOpen}>
+            <DialogTrigger asChild>
+              <Button disabled={isSubmitDisabled} size="lg" variant="primary-cta">
+                Submit
+              </Button>
+            </DialogTrigger>
+            <DialogContent a11yTitle="Active Poll Submit Button" className="max-w-l w-full">
+              <DialogBody>
+                <UserActionFormActionUnavailable />
+              </DialogBody>
+            </DialogContent>
+          </Dialog>
+        )
+      }
+      bypassCountryCheck={false}
+    >
+      <Button disabled={isSubmitDisabled} size="lg" variant="primary-cta">
+        Submit
+      </Button>
+    </LoginDialogWrapper>
+  )
```

Comment:
The button render repeats itself 3 times on this return statement, since it always render we could just remove the render from the return statement without changing behavior 
```suggestion
  const buttonRender = (
    <Button disabled={isSubmitDisabled} size="lg" variant="primary-cta">
      Submit
    </Button>
  )
  
  return (
    <LoginDialogWrapper
      authenticatedContent={
        isValid ? (
          buttonRender
        ) : (
          <Dialog {...dialogProps} onOpenChange={setDialogOpen}>
            <DialogTrigger asChild>
              {buttonRender}
            </DialogTrigger>
            <DialogContent a11yTitle="Active Poll Submit Button" className="max-w-l w-full">
              <DialogBody>
                <UserActionFormActionUnavailable />
              </DialogBody>
            </DialogContent>
          </Dialog>
        )
      }
      bypassCountryCheck={false}
    >
      {buttonRender}
    </LoginDialogWrapper>
  )
```